tests/smoke-tests/test-ethereum-db-mapping.ts

```diff
@@ -0,0 +1,40 @@
+import { expect } from "chai";
+import { describeSmokeSuite } from "../util/setup-smoke-tests";
+
+const debug = require("debug")("smoke:ethereum-mapping");
+
+const wssUrl = process.env.WSS_URL || null;
+const ethRpcUrl = process.env.HTTPS_URL || null;
+const relayWssUrl = process.env.RELAY_WSS_URL || null;
+
+// At rpc-level, `*ByNumber` requests always use the canonical block reference given by Substrate.
+// In the other hand `*ByHash` requests rely on data mapped in the frontier db.
+// We want to compare both to verify recent db data consistency and rpc impl across client versions.
+
+describeSmokeSuite(
+  `Ethereum secondary DB should contains valid data`,
+  { wssUrl, relayWssUrl, ethRpcUrl },
+  (context) => {
+    it("should get the same response payload on byNumber and byHash requests", async function () {
+      this.timeout(6_000_000); // 30 minutes
+      // As we are testing rpc-level functionality the height at which we access secondary db data
+      // is irrelevant. We can just select some arbitrary block numbers to verify block hashes.
+      const latestBlockNumber = await context.ethers.getBlockNumber();
+      // We asume we only want to run the test in a live chain.
+      if (latestBlockNumber > 10000) {
+        const checkPoint_1 = latestBlockNumber - 10;
+        const checkPoint_2 = latestBlockNumber - 100;
+        const checkPoint_3 = latestBlockNumber - 1000;
+        const checkPoint_4 = latestBlockNumber - 10000;
+
+        const blocks = [latestBlockNumber, checkPoint_1, checkPoint_2, checkPoint_3, checkPoint_4];
+
+        for (const block of blocks) {
+          const byNumber = await context.ethers.getBlock(block);
+          const byHash = await context.ethers.getBlock(byNumber.hash);
+          expect(byNumber).to.be.deep.eq(byHash);
```

Comment:
This test will halt on the first instance of an assertion failure, hiding if there are failures on  additional checkpoints. 

Could you instead the map the negative results to a list and then do an check that the list is empty at the end? 

That way you can print out the results of all the checkpoints - which will improve debugging if this happens live (as it'll show us if lots of blocks are broken or not).

Reply:
We can, but the thing is checkpoints heights are irrelevant for this test, they are added to go back "enough" so a recent client version transition is tested against. If some of the checkpoint fail, we don't really care which of them did, it just means a change was introduced in the most recent client version that was incorrect, either at rpc-level or a migration on the secondary db added inconsistencies.

Reply:
I think it is fine for this time, but Tim is right.
We often prefer to have all the failures collected and reported, instead of failing on the first one. The main reason being that if you have all the blocks failing or just 1, it can mean something different (maybe it was the runtime at that particular block that could have impacted the change, or it could be....). It adds additional information for debugging

Reply:
Just pushed your suggestions PTAL.